pkg/dr/health.go

```diff
@@ -0,0 +1,94 @@
+package dr
+
+import (
+	ctx "context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	rookv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type secretData struct {
+	Key     string `json:"key"`
+	MonHost string `json:"mon_host"`
+}
+
+func Health(context *k8sutil.Context, operatorNamespace, cephClusterNamespace string, args []string) {
+	fmt.Println("INFO: fetching the cephblockpools with mirroring enabled")
+	blockPoolList, err := context.RookClientset.CephV1().CephBlockPools(cephClusterNamespace).List(ctx.TODO(), v1.ListOptions{})
+	if err != nil {
+		log.Error(err)
+	}
+
+	var mirrorBlockPool rookv1.CephBlockPool
+	for _, blockPool := range blockPoolList.Items {
+		if blockPool.Spec.Mirroring.Enabled && blockPool.Spec.Mirroring.Peers != nil {
+			mirrorBlockPool = blockPool
+			fmt.Printf("found %q cephblockpool with mirroring enabled\n", mirrorBlockPool.Name)
+		}
+	}
+
+	if mirrorBlockPool.Name == "" {
+		fmt.Printf("DR is not confiqured, cephblockpool with mirroring enabled not found.\n")
+	}
+
+	secretDetail := mirrorBlockPool.Spec.Mirroring.Peers.SecretNames[0]
+
+	secretData, err := extractSecretName(context, operatorNamespace, cephClusterNamespace, secretDetail)
+	if err != nil {
+		fmt.Printf("failed to extract secret %s", secretDetail)
+		log.Fatal(err)
+	}
+
+	fmt.Println("running ceph status from peer cluster")
+
+	cephArgs := []string{"-s", "--mon-host", secretData.MonHost, "--id", "rbd-mirror-peer", "--key", secretData.Key}
+
+	if len(args) == 0 {
+		cephArgs = append(cephArgs, "--debug-ms", "0")
+	} else {
+		cephArgs = append(cephArgs, args...)
+	}
+
+	cephStatus := exec.RunCommandInOperatorPod(context, "ceph", cephArgs, operatorNamespace, cephClusterNamespace, false)
+	if cephStatus == "" {
+		fmt.Println("failed to get ceph status from peer cluster, please check for network issues between the clusters")
+	}
+	fmt.Println(cephStatus)
+
+	fmt.Println("running mirroring daemon health")
+
+	fmt.Println(exec.RunCommandInOperatorPod(context, "rbd", []string{"-p", mirrorBlockPool.Name, "mirror", "pool", "status"}, cephClusterNamespace, operatorNamespace, true))
+}
+
+func extractSecretName(context *k8sutil.Context, operatorNamespace, cephClusterNamespace, secretDetail string) (*secretData, error) {
+
+	secretName := strings.TrimSuffix(secretDetail, "]")
+	secretName = strings.TrimPrefix(secretName, "[")
+	// TODO: While testing Rook canary mirroring test, found that there is extra suffic `-config`
+	// in blockPool cr under mirroring which needs to be removed. Below is working but need to do RCA.
+	// secretName = strings.TrimSuffix(secretName, `-config`)
```

Comment:
The CI test patches the secret name to the cephBlockPool CR
`kubectl -n rook-ceph patch cephblockpool replicapool --type merge -p '{"spec":{"mirroring":{"peers": {"secretNames": ["pool-peer-token-replicapool-config"]}}}}'`

This name is being send as an argument to this method (`secretDetail`). I don't think we need the `TrimPrefix` method here.

Reply:
got it, removed.